### PR TITLE
Add client connection test coverage

### PR DESCRIPTION
**Pull request by jonmanahan**

### CODE REVIEW

File: app/src/test/java/waiter/ClientConnectionTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.io.IOException;
+
+public class ClientConnectionTest {
+
+    @Test
+    void readsMessage() throws IOException {
+        String input = "foo";
+        InputStreamerMock inputStreamerMock = new InputStreamerMock(input);
+        OutputStreamerMock outputStreamerMock = new OutputStreamerMock();
+        ClientConnection clientConnection = new ClientConnection(inputStreamerMock, outputStreamerMock);
```
Comment (mrfishball) [inline app/src/test/java/waiter/ClientConnectionTest.java:20]: What do you think about inlining this:
```java
String input = "foo";

ClientConnection clientConnection = new ClientConnection(
  new InputStreamerMock(input),
  new OutputStreamerMock()
);
```

File: app/src/test/java/waiter/ClientConnectionTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        assertEquals(readMessage.open(), input);
+    }
+
+    @Test
+    void writesMessage() throws IOException {
+        String input = "foo";
+        Message outputMessage = new Message(input);
+        InputStreamerMock inputStreamerMock = new InputStreamerMock(input);
+        OutputStreamerMock outputStreamerMock = new OutputStreamerMock();
+        ClientConnection clientConnection = new ClientConnection(inputStreamerMock, outputStreamerMock);
```
Comment (mrfishball) [inline app/src/test/java/waiter/ClientConnectionTest.java:31]: Similarly, consider inlining here.

File: app/src/test/java/waiter/ClientConnectionTest.java (hunk 1 above)
Comment (jonmanahan) [reply app/src/test/java/waiter/ClientConnectionTest.java:20]: Hmmm, I think this is a good idea, it doesn't really hurt readability and it makes it more concise, should I reflect this in `Listener` as well?

File: app/src/test/java/waiter/ClientConnectionTest.java (hunk 2 above)
Comment (jonmanahan) [reply app/src/test/java/waiter/ClientConnectionTest.java:31]: That makes sense, ignore the second part of my comment above 😂

**Review by mrfishball — APPROVED** [review]
Left a non blocking comment. Looks good!

File: app/src/test/java/waiter/ClientConnectionTest.java (hunk 1 above)
Comment (mrfishball) [reply app/src/test/java/waiter/ClientConnectionTest.java:20]: Yea sounds good to me!

File: app/src/test/java/waiter/ClientConnectionTest.java (hunk 2 above)
Comment (mrfishball) [reply app/src/test/java/waiter/ClientConnectionTest.java:31]: how about this:
```java
String input = "foo";
OutputStreamerMock outputStreamerMock = new OutputStreamerMock();

ClientConnection clientConnection = new ClientConnection(
  InputStreamerMock(foo),
  outputStreamerMock
);

assertEquals(outputStreamerMock.writtenOutput, input);
```

I feel like you don't really line 28, I could be wrong though.

File: app/src/main/java/waiter/OutputStreamer/OutputStreamer.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class OutputStreamer implements Writer {
+
+    private final Socket socket;
+
+    public OutputStreamer(Socket socket) {
+        this.socket = socket;
+    }
+
+    public void writeLine(String toClient) throws IOException {
+        PrintStream printStream = new PrintStream( this.socket.getOutputStream() );
```
Comment (mrfishball) [inline app/src/main/java/waiter/OutputStreamer/OutputStreamer.java:17]: Is the beginning and trailing spacing here intentional? `... new PrintStream( this.socket.getOutputStream() );`

**Pull request merged**